### PR TITLE
feat: Added Disconnect command for every vCluster-related context

### PR DESCRIPTION
**Pull request by ThomasK33**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves ENG-3727

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where vcluster ...

**What else do we need to know?**

### CODE REVIEW

File: cmd/vclusterctl/cmd/platform/connect/vcluster.go [hunk 1]
```diff
@@ -1,121 +0,0 @@
-package connect
```
Comment (johannesfrey) [inline cmd/vclusterctl/cmd/platform/connect/vcluster.go:1]: Why is this removed? I thought `vcluster connect` is not the same as`vcluster platform connect vcluster`, because in `vcluster connect` you don't have those flags: https://github.com/loft-sh/vcluster/blob/main/cmd/vclusterctl/cmd/platform/connect/vcluster.go#L23-L27
Or should they rather be moved over to the `ConnectOptions`?

Comment (ThomasK33) [reply cmd/vclusterctl/cmd/platform/connect/vcluster.go:1]: Yes, those should be part of the `vcluster connect` command.

Having two commands to do the same thing only adds to confusion about when to use which command.
Given that the `vcluster create` commands directly creates an instance in the platform, the `vcluster connect` command should be the one to also connect to it, with all the flags necessary.

**Review by johannesfrey — APPROVED** [review]
